File: src/main/java/brain/suppliers/ConstSupplier.java
package brain.suppliers;

import java.util.Arrays;

public class ConstSupplier implements ValueFiller {

	private final float value;

	public ConstSupplier(float value) {
		this.value = value;
	}

	@Override
	public float supply(int count, int x, int y) {
		return value;
	}

	@Override
	public void fill(float[] array) {
		Arrays.fill(array, value);
	}
}

File: src/main/java/brain/suppliers/ConstSupplier.kt
package brain.suppliers

open class ConstSupplier(val const: Float) : ValueSupplier {
	override fun fill(array: FloatArray) {
		array.fill(const)
	}

	override fun create(size: Int): FloatArray {
		return FloatArray(size) { const }
	}
}